pkgs/development/embedded/stm32/stm32cubemx/default.nix

```diff
@@ -1,84 +1,115 @@
+      chmod +rx $out/opt/STM32CubeMX/STM32CubeMX
 
-    icotool --extract $out/opt/STM32CubeMX/help/${iconame}.ico
-    fdupes -dN . > /dev/null
-    ls
-    for size in 16 24 32 48 64 128 256; do
-      mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
-      if [ $size -eq 256 ]; then
-        mv ${iconame}_*_"$size"x"$size"x32.png \
-          $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
-      else
-        convert -resize "$size"x"$size" ${iconame}_*_256x256x32.png \
-          $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
-      fi
-    done;
+      cat << EOF > $out/bin/${pname}
+      #!${stdenvNoCC.shell}
+      ${jdk17}/bin/java -jar $out/opt/STM32CubeMX/STM32CubeMX
+      EOF
+      chmod +x $out/bin/${pname}
 
-    cp ${desktopItem}/share/applications/*.desktop $out/share/applications
-  '';
+      icotool --extract $out/opt/STM32CubeMX/help/${iconame}.ico
+      fdupes -dN . > /dev/null
+      ls
+      for size in 16 24 32 48 64 128 256; do
+        mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
+        if [ $size -eq 256 ]; then
+          mv ${iconame}_*_"$size"x"$size"x32.png \
+            $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
+        else
+          convert -resize "$size"x"$size" ${iconame}_*_256x256x32.png \
+            $out/share/icons/hicolor/"$size"x"$size"/apps/${pname}.png
+        fi
+      done;
 
-  meta = with lib; {
-    description = "A graphical tool for configuring STM32 microcontrollers and microprocessors";
-    longDescription = ''
-      A graphical tool that allows a very easy configuration of STM32
-      microcontrollers and microprocessors, as well as the generation of the
-      corresponding initialization C code for the Arm® Cortex®-M core or a
-      partial Linux® Device Tree for Arm® Cortex®-A core), through a
-      step-by-step process.
+      cp ${desktopItem}/share/applications/*.desktop $out/share/applications
     '';
-    homepage = "https://www.st.com/en/development-tools/stm32cubemx.html";
-    sourceProvenance = with sourceTypes; [ binaryBytecode ];
-    license = licenses.unfree;
-    maintainers = with maintainers; [ angaz wucke13 ];
-    platforms = platforms.all;
+
+    meta = with lib; {
+      description = "A graphical tool for configuring STM32 microcontrollers and microprocessors";
+      longDescription = ''
+        A graphical tool that allows a very easy configuration of STM32
+        microcontrollers and microprocessors, as well as the generation of the
+        corresponding initialization C code for the Arm® Cortex®-M core or a
+        partial Linux® Device Tree for Arm® Cortex®-A core), through a
+        step-by-step process.
+      '';
+      homepage = "https://www.st.com/en/development-tools/stm32cubemx.html";
+      sourceProvenance = with sourceTypes; [ binaryBytecode ];
+      license = licenses.unfree;
+      maintainers = with maintainers; [ angaz wucke13 ];
+      platforms = platforms.all;
```

Comment:
I believe it is a bit misleading, since Java does not run in all platforms.
But I don't think it will be a serious problem.

Reply:
That was a bit out of scope since I focused on bugfix.
Though even more we have reports that the package doesn't work on aarm64 because of upstream issues but I think we should leave platforms revision for later PR if that is fine.

Reply:
OK.

Reply:
The app is only released for x86. I think this is a simple thing to fix, and can be done in this PR. It should be good for users to know it won't work on other platforms.

Reply:
Missed this.
OK, I changed it to x86_64-linux only.